### PR TITLE
Add consistent_relative_to_receiver to FirstParameterIndentation

### PR DESCRIPTION
**Pull request by jfelchner**
There are multiple valid issues with people complaining about how `FirstParameterIndentation`'s chosen correction is not very readable.

This pull request introduces a new style to the cop called `consistent_relative_to_receiver` which will always align the first parameter relative to the receiver/parent node instead of looking at the line's indentation.

Examples:

### Actual

```ruby
MyClass.my_method(a_hash.merge(
  hello: :world,
  some: :hash,
  goes: :here
), other_arg)
```

### vs Expected

```ruby
MyClass.my_method(a_hash.merge(
                    hello: :world,
                    some: :hash,
                    goes: :here
                  ), other_arg)
```

### Actual

```ruby
good3 = foo.methA(
  :arg1,
  :arg2,
  options: :hash,
)
          .methB(
            :arg1,
            :arg2,
          )
          .methC
```

### vs Expected

```ruby
good3 = foo.methA(
          :arg1,
          :arg2,
          options: :hash,
        )
        .methB(
          :arg1,
          :arg2,
        )
        .methC
```

### Actual

```ruby
price = default_price * NightsCalculator.new(
  start_date,
  end_date).nights_count
```

### vs Expected

```ruby
price = default_price * NightsCalculator.new(
                          start_date,
                          end_date).nights_count
```

References: #5650, #4308, #3204, #2625

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by jfelchner** [comment]
@bbatsov @jonas054 @rrosenblum @Drenmi if one of you could help me out, I need some assistance getting this over the finish line.

There are two tests failing on this branch.  The assertions are correct.  I believe that what's happening is the `AlignmentCorrector` is only looking at the top-most node for the indention level it should be correcting the alignment to, but in order for the tests to pass, it requires aligning based on multiple levels.

I just don't know enough about some of the Rubocop helpers/internals to know what the best approach to take is here.  If someone could help me out with this, between this PR and #5698 we should be able to close out around 8-10 issues. 👍

**Comment by bbatsov** [comment]
```
good3 = foo.methA(
  :arg1,
  :arg2,
  options: :hash,
)
          .methB(
            :arg1,
            :arg2,
          )
          .methC
```

I'd say that's a bug in whatever style. :-)

I'm overloaded with work, but I'll take a look at this when I can if someone else doesn't beat me to it. We can also ask for help @garettarrowood, @pocke and @koic.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### New features
 
 * [#5597](https://github.com/bbatsov/rubocop/pull/5597): Add new `Rails/HttpStatus` cop. ([@anthony-robin][])
+* [#5699](https://github.com/bbatsov/rubocop/pull/5699): Add `consistent_relative_to_receiver` style option to `Layout/FirstParameterIndentation` ([@jfelchner][])
```
Comment (garettarrowood) [inline CHANGELOG.md]: One failure is that this entry doesn't end with punctation. Adding a period will resolve it.

Comment (garettarrowood) [reply CHANGELOG.md]: I will see if I can give the meaningful failures a closer look this weekend. I've also been swamped and haven't been able to give `rubocop` the attention I'd like to over the past couple months.

Comment (jfelchner) [reply CHANGELOG.md]: I was speaking more about the test failures.

**Comment by jonas054** [comment]
@jfelchner What's happening in the failing tests is that the auto-correction of one line changes how another line needs to be aligned. This is something that we solve in RuboCop by running cops repeatedly until no more correctable offenses are found. But we can't do it within a single run of a cop.

So I recommend you change these examples to one offense only.

**Comment by jfelchner** [comment]
@bbatsov rebased onto `master` and force pushed with requested updates.  Everything should be good to go (assuming all the specs pass).  I even did a big update on the doc comments for the cop to show examples of correct formatting for all the different styles the cop accepts.

**Comment by jfelchner** [comment]
@bbatsov rebased to fix conflicts

**Comment by jfelchner** [comment]
@bbatsov did you want additional changes here before merging?

**Comment by bbatsov** [comment]
Nope. Thanks! 👍

**Pull request merged**